tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionTrackManager.cs

```diff
@@ -759,6 +757,40 @@ private static bool ShouldSkip(MethodBase method)
 
         private static void SetDiagnosticTag(Span span, string exceptionPhase, int exceptionHash)
         {
+            if (exceptionPhase is ExceptionReplayDiagnosticTagNames.CachedInvalidatedExceptionCase or ExceptionReplayDiagnosticTagNames.CachedDoneExceptionCase or ExceptionReplayDiagnosticTagNames.NonCachedDoneExceptionCase)
+            {
+                return;
+            }
+
+            var noCaptureReason = string.Empty;
+
+            switch (exceptionPhase)
+            {
+                case ExceptionReplayDiagnosticTagNames.Eligible:
+                    break;
+                case ExceptionReplayDiagnosticTagNames.NoCustomerFrames or ExceptionReplayDiagnosticTagNames.NoFramesToInstrument:
+                    noCaptureReason = NoCaptureReason.OnlyThirdPartyCode;
+                    break;
+                case ExceptionReplayDiagnosticTagNames.InvalidatedCase or ExceptionReplayDiagnosticTagNames.InvalidatedExceptionCase:
+                    noCaptureReason = NoCaptureReason.InstrumentationFailure;
+                    break;
+                case ExceptionReplayDiagnosticTagNames.NonSupportedExceptionType:
+                    noCaptureReason = NoCaptureReason.NonSupportedExceptionType;
+                    break;
+                case ExceptionReplayDiagnosticTagNames.NotEligible or ExceptionReplayDiagnosticTagNames.NewCase:
+                    noCaptureReason = NoCaptureReason.FirstOccurrence;
+                    break;
+                default:
+                    noCaptureReason = NoCaptureReason.GeneralError;
+                    break;
+            }
```

Comment:
nit: switch expressions FTW 😄 
```suggestion
            var noCaptureReason = exceptionPhase switch
            {
                ExceptionReplayDiagnosticTagNames.Eligible => string.Empty,
                ExceptionReplayDiagnosticTagNames.NoCustomerFrames or ExceptionReplayDiagnosticTagNames.NoFramesToInstrument => NoCaptureReason.OnlyThirdPartyCode,
                ExceptionReplayDiagnosticTagNames.InvalidatedCase or ExceptionReplayDiagnosticTagNames.InvalidatedExceptionCase => NoCaptureReason.InstrumentationFailure,
                ExceptionReplayDiagnosticTagNames.NonSupportedExceptionType => NoCaptureReason.NonSupportedExceptionType,
                ExceptionReplayDiagnosticTagNames.NotEligible or ExceptionReplayDiagnosticTagNames.NewCase => NoCaptureReason.FirstOccurrence,
                _ => NoCaptureReason.GeneralError,
            };
```

Reply:
Done in ea7b1b579ca5d4d6d89c59136d6bbe72985d3cfb.